charts/tekton-pipelines/templates/common/tasks/sentry-release.yaml

```diff
@@ -0,0 +1,82 @@
+{{ if .Values.sentry.enabled }}
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: sentry-release
+  namespace: {{ .Release.Namespace }}
+
+spec:
+  description: >-
+    The task prepares and deploys sentry release for projects, which exist in Sentry.
+
+  workspaces:
+    - name: source
+
+  resources:
+    inputs:
+      - name: app
+        type: git
+
+  params:
+    - name: project
+      type: string
+      description: name of the project, which component is deployed
+
+    - name: component
+      type: string
+      description: name of the application component, which is deployed
+
+    - name: environment
+      type: string
+      description: environment name of the app being built, i.e. dev/staging/prod
+
+    - name: sourcemaps_dir
+      type: string
+      description: name of the dir where frontend sourcemaps would be stored in workspace
+      default: "sourcemaps"
+
+  stepTemplate:
+    env:
+      - name: SENTRY_AUTH_TOKEN
+        valueFrom:
+          secretKeyRef:
+            name: {{ .Values.sentry.authTokenSecret }}
+            key: {{ .Values.sentry.authTokenSecretKey }}
+      - name: SENTRY_ORG
+        value: {{ .Values.sentry.org }}
+      - name: SENTRY_URL
+        value: {{ .Values.sentry.url }}
+
+  steps:
+    - name: release
+      image: {{ .Values.images.sentry_cli }}
+      imagePullPolicy: {{ .Values.imagePullPolicy }}
+      workingDir: $(resources.inputs.app.path)
+      script: |
+        #!/usr/bin/env sh
+        set +x
+
+        PROJECT_NAME="$(params.project)-$(params.component)"
+        PROJECT=$(sentry-cli projects list | awk -v col=4 '{print $col}' | grep ^${PROJECT_NAME}$)
+        if [[ -z "${PROJECT}" ]]; then
+          echo "Didn't find '${PROJECT_NAME}' project in Sentry. Skip this step."
+          exit 0
+        else
+          echo "Found '${PROJECT_NAME}' project in Sentry, proceed with release."
+        fi
+
+        VERSION=$(sentry-cli releases propose-version)
```

Comment:
so how is this working? What VERSION var would contain after calling this CLI?
does it take tag name or sha1 commit ID into consideration? Something else?

Reply:
added more comments

Reply:
how does this work for solution with tags, like git-version (and mods Misha made for Anton) that is tag based?
I.e. for them it would be easier to see as release/x.y.z rather than 7e734134fb23497e68976784f7eaa106c4a8f3a8?

Reply:
Original `sentry-cli releases propose-version` doesn't work with tags, it generates version based on last commit hash only. We can configure our own version based on tags or add any other sensible approach to name releases better than just last commit hash (so we will need to implement custom bash solution to prepare better release name instead of `sentry-cli releases propose-version` usage). By the way, our current Tekton releasing workflow doesn't have a feature of adding tags to `main` or `staging` branches, we just release from branches itself, not from tags. So we will need to make updates to our Tekton pipelines for it too, to be able to add tags durning deployment.

But I am not sure whether we really need it that much now, it only influences releases readability (https://sentry.saritasa.rocks/organizations/saritasa/releases/?environment=dev), but basically functionality stays the same, so we will have all commits between `last` released commit and current one. Also currently we have `env` filtration in Releases, so we would be able to separate releases by envs.

So, I suggest to leave as it is for now and maybe later when we will have dev + staging + prod experience with Sentry we can consider its improvements.